### PR TITLE
LIIKUNTA-581 | update unified search related query arguments

### PR DESCRIPTION
**Pull request by karisal-anders**
## Description

**NOTE**:
- **Review environment does not work** correctly **because** federation router is not up to date. **Review environment does not use review federation router** (See [DevOps pipeline diagram](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HELFI/pages/8265859073/Liikunta+-+DevOps+pipelines)) but test federation router which does not get the updated supergraph for the federation router from this PR. **Tested locally**

### fix: update unified search query arguments

unified search query arguments:
 - remove:
   - before
   - last
   - administrativeDivisionId
   - ontologyWordIds
   - ontologyTreeId
   - ontologyTreeIds
   - ontologyTreeIdsOrSet2

 - add:
   - ontologyTreeIdOrSets
   - ontologyWordIdOrSets

 - migrate:
   - administrativeDivisionId -> administrativeDivisionIds
   - ontologyWordIds -> ontologyWordIdOrSets
   - ontologyTreeId -> ontologyTreeIdOrSets
   - ontologyTreeIds -> ontologyTreeIdOrSets
   - ontologyTreeIdsOrSet2 -> ontologyTreeIdOrSets

also:
 - remove useUnifiedSearch and related files as unnecessary

refs LIIKUNTA-581
refs US-115 (breaking changes to unified-search graphql query schema)

### fix: rename "q" argument to "text" argument in unified & venue search

refs LIIKUNTA-581
refs US-115 (changes to unified-search)

### fix: rename orderedByAccessibilityShortcoming to accessibilityShortcomingFor

refs LIIKUNTA-581
refs US-115 (changed unified-search graphql schema)

### chore(proxies): update unified-search subgraph and supergraph

refs LIIKUNTA-581
refs US-115 (changes made to unified-search's graphql schema)

### Closes

[LIIKUNTA-581](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-581)

### Related

[US-115](https://helsinkisolutionoffice.atlassian.net/browse/US-115)

### Manual testing

- Tested locally with a local events-graphql-federation router also running locally
  - Landing page search worked with Finnish + "uimahalli" and English + "swimming hall" searches
  - One of the landing page sports category shortcuts worked
  - Venue search worked at least superficially, no toast errors
  - packages/components tests passed locally
  - apps/sports-helsinki tests passed locally

## Additional notes

[LIIKUNTA-581]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-581?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-events-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-hobbies-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr525.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-sports-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-hobbies-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr525.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://liikunta-pr525.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://harrastukset-pr525.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-hobbies-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Federation Router branch is deployed to platta: https://events-graphql-federation-events-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tapahtumat-pr525.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr525.dev.hel.ninja** :rocket::rocket::rocket:

**Review by nikomakela — APPROVED** [review]
There are quite a lot of code changes, but just by reading them through, they look OK. I did try the search view and it's not working 
image
Is it so because the unified search is not deployed or is there something else?

File: packages/components/src/components/domain/unifiedSearch/graphql/unifiedSearchMap.query.ts [hunk 1]
```diff
@@ -17,14 +17,14 @@ export const SEARCH_MAP_QUERY = gql`
     $orderByName: OrderByName
   ) {
     unifiedSearch(
-      q: $q
-      index: "location"
```
Comment (nikomakela) [inline packages/components/src/components/domain/unifiedSearch/graphql/unifiedSearchMap.query.ts:21]: what is this change? What does it do?

Comment (nikomakela) [reply packages/components/src/components/domain/unifiedSearch/graphql/unifiedSearchMap.query.ts:21]: what I meant was that the index earlier got "", but now does not

Comment (karisal-anders) [reply packages/components/src/components/domain/unifiedSearch/graphql/unifiedSearchMap.query.ts:21]: The `index`'s type changed from `String` to `enum UnifiedSearchIndex`:
```graphql
  enum UnifiedSearchIndex {
    administrative_division
    helsinki_common_administrative_division
    ontology_tree
    ontology_word
    event
    location
  }
```
That's what necessitated the removal of the quotes around "location" here.

**Comment by karisal-anders** [comment]
> I did try the search view and it's not working. Is it so because the unified search is not deployed or is there something else?

Review environment does not work correctly because federation router is not up to date. Review environment does not use review federation router (See [DevOps pipeline diagram](https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HELFI/pages/8265859073/Liikunta+-+DevOps+pipelines)) but test federation router which does not get the updated supergraph for the federation router from this PR.

**Pull request merged**